src/libbson/src/bson/bson-utf8.c

```diff
@@ -230,6 +230,118 @@ bson_utf8_validate (const char *utf8, /* IN */
    return true;
 }
 
+/*
+ *--------------------------------------------------------------------------
+ *
+ * _is_special_char --
+ *
+ *       Uses a bit mask to check if a character requires special formatting
+ *       or not. Called from bson_utf8_escape_for_json.
+ *
+ * Parameters:
+ *       @c: An unsigned char c.
+ *
+ * Returns:
+ *       true if @c requires special formatting. otherwise false.
+ *
+ * Side effects:
+ *       None.
+ *
+ *--------------------------------------------------------------------------
+ */
+
+static BSON_INLINE bool
+_is_special_char (unsigned char c)
+{
+   /*
+   C++ equivalent:
+   std::bitset<256> charmap = [...]
+   return charmap[c];
+   */
+   static const bson_unichar_t charmap[8] = {0xffffffff, // control characters
+                                             0x00000004, // double quote "
+                                             0x10000000, // backslash
+                                             0x00000000,
+                                             0xffffffff,
+                                             0xffffffff,
+                                             0xffffffff,
+                                             0xffffffff}; // non-ASCII
+   const int int_index = c / (sizeof (bson_unichar_t) * 8);
+   const int bit_index = c & (sizeof (bson_unichar_t) * 8 - 1);
+   return (charmap[int_index] >> bit_index) & 1;
```

Comment:
```suggestion
   const int int_index = ((int) c) / ((int) sizeof (bson_unichar_t) * 8);
   const int bit_index = ((int) c) & ((int) sizeof (bson_unichar_t) * 8 - 1);
   return ((charmap[int_index] >> bit_index) & ((bson_unichar_t) 1)) != 0u;
```
Signedness and type consistency.